pco/src/latent_batch_dissector.rs

```diff
@@ -20,10 +20,17 @@ pub struct LatentBatchDissector<'a, L: Latent> {
 
 impl<'a, L: Latent> LatentBatchDissector<'a, L> {
   pub fn new(table: &'a CompressionTable<L>, encoder: &'a ans::Encoder) -> Self {
+    // We initialize the scratch buffer for bin lowers carefully to enable
+    // a shortcut when there's only one bin.
```

Comment:
A tricky thing about changes like this is that the places in the codebase that have to know about this shortcut are not all right next to each other.  Is there a spec/docstring to which we can add a representation invariant along the lines of "If there are 0 or 1 bins, then the scratch buffer `lower_scratch` is always populated.  If there are more than one bin, then `lower_scratch` is not correctly populated until such-and-such methods have been called"?  It might not be that clear-cut, but I wanted to raise the question in case there is a way.

Reply:
I'm not sure how that would be done with code without some annoying complexity. These functions aren't user-facing, so there are no doc comments, but hopefully these regular comments are sufficient?